clients/js/src/unwrap.ts

```diff
@@ -0,0 +1,217 @@
+import {
+  Address,
+  appendTransactionMessageInstructions,
+  CompilableTransactionMessage,
+  createTransactionMessage,
+  fetchEncodedAccount,
+  GetAccountInfoApi,
+  pipe,
+  Rpc,
+  setTransactionMessageFeePayerSigner,
+  setTransactionMessageLifetimeUsingBlockhash,
+  TransactionMessageWithBlockhashLifetime,
+  TransactionSigner,
+} from '@solana/kit';
+import { getTokenDecoder } from '@solana-program/token-2022';
+import { findWrappedMintAuthorityPda, getUnwrapInstruction, UnwrapInput } from './generated';
+import { Blockhash } from '@solana/rpc-types';
+import { getMintFromTokenAccount, getOwnerFromAccount } from './utilities';
+
+export interface SingleSignerUnwrapArgs {
+  rpc: Rpc<GetAccountInfoApi>;
+  blockhash: {
+    blockhash: Blockhash;
+    lastValidBlockHeight: bigint;
+  };
+  payer: TransactionSigner; // Fee payer and default transfer authority
+  wrappedTokenAccount: Address;
+  unwrappedEscrow: Address;
+  amount: bigint | number;
+  recipientUnwrappedToken: Address;
+  // Optional arguments below (will be derived/defaulted if not provided)
+  transferAuthority?: Address | TransactionSigner; // Defaults to payer
+  unwrappedMint?: Address; // Will derive from unwrappedEscrow if not provided
+  wrappedTokenProgram?: Address; // Will derive from wrappedTokenAccount if not provided
+  unwrappedTokenProgram?: Address; // Will derive from unwrappedEscrow if not provided
+}
```

Comment:
Optional for a follow up PR - it might be worth using TSDoc here: https://tsdoc.org

The benefit would be that this param documentation will show inline in the editor for someone creating this, while these comments typically won't

Reply:
Ah, good to know!